storage/libradb/src/lib.rs

```diff
@@ -687,18 +687,20 @@ impl DbReader for LibraDB {
             .with_label_values(&["get_account_state_with_proof"])
             .start_timer();
 
-        if version <= ledger_version {
-            warn!(
-                "The queried version {} should be equal to or older than ledger version {}.",
-                version, ledger_version
-            );
-        }
-        let latest_version = self.get_latest_version()?;
-        if ledger_version <= latest_version {
-            warn!(
-                "The ledger version {} is greater than the latest version currently in ledger: {}",
-                ledger_version, latest_version
-            )
+        ensure!(
+            version <= ledger_version,
+            "The queried version {} should be equal to or older than ledger version {}.",
+            version,
+            ledger_version
+        );
+        {
+            let latest_version = self.get_latest_version()?;
+            if ledger_version > latest_version {
+                warn!(
```

Comment:
would be good to start leveraging structured logs in storage

would this result in a client breaking change? I guess kinda... but perhaps they were using it in ... poor taste

also I would caution that if this interface is exposed to clients, we probably do not want warn logs, because it can be used to dos our logging backend :/

Reply:
That's a great argument. Hold on.

Reply:
Realized there's already a "cheating" API used internally to ignore the ledger_version check. will squash and update the commit info  before landing.